ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts

```diff
@@ -131,21 +132,19 @@ export class ParticipantsListComponent implements OnInit {
   }
 
   onEditActivity(participantGuid: string, activityToEdit: ActivityInstance): void {
+    this.isPageBusy = true;
     this.setParticipant(participantGuid);
-
-    this.activityService
-      .createInstance(this.config.studyGuid, activityToEdit.activityCode)
-      .pipe(take(1))
-      .subscribe(activity => {
-        this.setCurrentActivity(activity as ActivityInstance);
-        this.redirectToSurvey(activity.instanceGuid);
-      });
+    this.setCurrentActivity(activityToEdit, false);
+    this.redirectToSurvey(activityToEdit.instanceGuid);
+    this.isPageBusy = false;
   }
 
```

Comment:
Creating activity instance here makes no sense to me. As you see, the method already receives an instance for editing, so why do we  need to create one?
Maybe there is a hidden reason? Take a look at JIRA ticket https://broadinstitute.atlassian.net/browse/DDP-8060.
My change was to get rid of the createInstance and just redirect to a web page where it can be edited.

Reply:
Hi Grigol. Yes. We have had that kind of interaction where a new activity is created, copying existing fields. And that seems to be what is happening. Current implementation, though, really causes a lot of problems.
On the other hand, I am not seeing that the old activity is reachable or viewable by users in any way (maybe from DSM?).

Reply:
I was suspicious that removing that new instance creation functionality might break activity edit functionality but looks like it doesn't. Edit still works just fine, it saves changed fields and I can see them changed, so I'm not sure why they had additional instance creation in the first place, maybe it's a remnant of copy-pasted code? Who knows, anyway I think your changes are legit